src/v35.js

```diff
@@ -1,12 +1,53 @@
 import bytesToUuid from './bytesToUuid.js';
 
+function hexSymToDecNum(n) {
+  // ------ 0 -------- 9
+  if (n >= 48 && n <= 57) {
+    return n - 48;
+  }
+  // ----------- A -------- F
+  else if (n >= 65 && n <= 70) {
+    return n - 55;
+  }
+  // ----------- a -------- f
+  else if (n >= 97 && n <= 102) {
+    return n - 87;
+  }
+
+  return -1;
+}
+
+/**
+ * Validate and parse UUID to bytes array
+ */
 function uuidToBytes(uuid) {
-  // Note: We assume we're being passed a valid uuid string
   const bytes = [];
 
-  uuid.replace(/[a-fA-F0-9]{2}/g, function (hex) {
-    bytes.push(parseInt(hex, 16));
-  });
+  if (uuid.length === 36) {
+    for (let i = 0; i < uuid.length; ++i) {
```

Comment:
Why not `i+=2` instead of `++i` here and another `++i` below?

Reply:
@ctavan because not in all cases it is necessary to increase by 2. If we find a `-`, then increase only by 1.
https://github.com/uuidjs/uuid/pull/439/files#diff-4c3fa2fbfa9d4b899d8be4b79ae6907fR33

Reply:
Right…